### ISSUE TITLE
Allow special characters for shape labels 

### ISSUE THREAD

**Issue by thewilly** [weso/shex-lite#20]
ShEx is more flexible when it comes to allowed characters in labels. For example, something like this will work:
```ttl
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
PREFIX sch-ema: <http://schema.org/>

<User> {
 sch-ema:1n-ame  xsd:string ;
 sch-ema:knows @<User>*
}
```

The same happens in ttl for example. I think it is worth following the already established conventions, but I would also like to receive your feedback on this topic.

_Originally posted by @alejgh in https://github.com/weso/shex-lite/pull/19_

**Comment by labra** [comment]
In my opinion the more close you work to Turtle/SPARQL/ShEx lexical rules, the better. To that end, the you can do is usually to copy the lexical rules from: http://shex.io/shex-semantics/#shexc

Notice that the rules are numbered and that those numbers are usually the same numbers as the ones in [Turtle](https://www.w3.org/TR/turtle/#sec-grammar) or [SPARQL](https://www.w3.org/TR/sparql11-query/#sparqlGrammar) grammars.

**Comment by thewilly** [comment]
> 
> 
> In my opinion the more close you work to Turtle/SPARQL/ShEx lexical rules, the better. To that end, the best you can do is usually to copy the lexical rules from: http://shex.io/shex-semantics/#shexc
> 
> Notice that the rules are numbered and that those numbers are usually the same numbers as the ones in [Turtle](https://www.w3.org/TR/turtle/#sec-grammar) or [SPARQL](https://www.w3.org/TR/sparql11-query/#sparqlGrammar) grammars.

Yes I'm trying to follow as much as possible meanwhile reducing the complexity and size of the generated parse tree. In this case the error was due to anmistake while translating the rule from the opficial specification as I used another rule that had a versy similar-confusing-name.

It is fixed at #19.
